### PR TITLE
Unit Testing with Coverage 

### PR DESCRIPTION
**Pull request by mohammadSalameh06**
This code includes comprehensive unit tests that cover all classes, excluding the Main object, which has been excluded from the build.sbt configuration. Additionally, a coverage report is generated as part of the build process, with a threshold set at 80%. The code consistently achieves coverage between 88% and 90%. Each class includes brief comments that explain the purpose of each function.

image

### CODE REVIEW

**Comment by abdalrahmanSrour** [comment]
Please update the title to have what are you changing in this PR, also you should add description to it with more details and a screen shot with coverage report.

File: .bsp/sbt.json [hunk 1]
```diff
@@ -1 +1 @@
-{"name":"sbt","version":"1.10.0","bspVersion":"2.1.0-M1","languages":["scala"],"argv":["C:\\Users\\user\\AppData\\Local\\Coursier\\cache\\arc\\https\\github.com\\adoptium\\temurin17-binaries\\releases\\download\\jdk-17%252B35\\OpenJDK17-jdk_x64_windows_hotspot_17_35.zip\\jdk-17+35/bin/java","-Xms100m","-Xmx100m","-classpath","C:/Users/user/AppData/Roaming/JetBrains/IdeaIC2024.1/plugins/Scala/launcher/sbt-launch.jar","-Dsbt.script=C:\\Users\\user\\AppData\\Local\\Coursier\\data\\bin\\sbt.bat","xsbt.boot.Boot","-bsp"]}
```
Comment (abdalrahmanSrour) [inline .bsp/sbt.json:1]: I think this file should be ignored, also add `.idea` folder and `.bsp` folder.

File: build.sbt [hunk 2]
```diff
@@ -1,18 +1,26 @@
...
+      "org.mockito" % "mockito-core" % "3.12.4" % Test
+  )
 
 )
+
+// Scoverage settings
+coverageEnabled := true
+coverageFailOnMinimum := true // Fail build if coverage is below minimum
+coverageHighlighting := true // Enable code highlighting in the coverage report
+coverageExcludedFiles := "ae.network.migration.pipeline.processExecute.MainTest" // Exclude test files
```
Comment (abdalrahmanSrour) [inline build.sbt]: You should exclude main not the test file.

File: src/main/scala/ae/network/migration/pipeline/parser/ParserJson.scala [hunk 3]
```diff
@@ -1,53 +1,52 @@
...
+  package ae.network.migration.pipeline.parser
 
-import ae.network.migration.pipeline.models.{Job, Pipeline, Stage}
-import io.circe.{Decoder, ParsingFailure}
-import io.circe.generic.semiauto.deriveDecoder
-import io.circe.jawn.decode
+  import ae.network.migration.pipeline.models.{Job, Pipeline, Stage}
+  import io.circe.{Decoder, ParsingFailure}
+  import io.circe.generic.semiauto.deriveDecoder
+  import io.circe.jawn.decode
```
Comment (abdalrahmanSrour) [inline src/main/scala/ae/network/migration/pipeline/parser/ParserJson.scala]: Please remove extra spaces before `import` keyword

File: src/main/scala/ae/network/migration/pipeline/processExcute/PipelineProcessor.scala [hunk 4]
```diff
@@ -1,11 +1,11 @@
-package ae.network.migration.pipeline.processExecute
+  package ae.network.migration.pipeline.processExecute
 
-import ae.network.migration.pipeline.models.Pipeline
-import scala.concurrent.{Future, ExecutionContext}
+  import ae.network.migration.pipeline.models.Pipeline
+  import scala.concurrent.{Future, ExecutionContext}
```
Comment (abdalrahmanSrour) [inline src/main/scala/ae/network/migration/pipeline/processExcute/PipelineProcessor.scala]: Same as above.

File: src/test/scala/ae/network/migration/pipeline/MainTest.scala [hunk 5]
```diff
@@ -0,0 +1,92 @@
+//package ae.network.migration.pipeline
```
Comment (abdalrahmanSrour) [inline src/test/scala/ae/network/migration/pipeline/MainTest.scala]: Delete this file.

**Review by abdalrahmanSrour — CHANGES_REQUESTED** [review]

File: build.sbt (hunk 2 above)
Comment (mohammadSalameh06) [reply build.sbt]: Done

File: src/main/scala/ae/network/migration/pipeline/parser/ParserJson.scala (hunk 3 above)
Comment (mohammadSalameh06) [reply src/main/scala/ae/network/migration/pipeline/parser/ParserJson.scala]: Done

File: src/test/scala/ae/network/migration/pipeline/MainTest.scala (hunk 5 above)
Comment (mohammadSalameh06) [reply src/test/scala/ae/network/migration/pipeline/MainTest.scala]: done

**Comment by mohammadSalameh06** [comment]
@abdalrahmanSrour all the requested changes are done , can you please check if the **bsp and idea**  are deleted or not.
thanks

**Review by abdalrahmanSrour — APPROVED** [review]

**Pull request merged**